src/main/java/ua/su/service/impl/FindSomethingInClinic2.java

```diff
@@ -0,0 +1,32 @@
+package ua.su.service.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import ua.su.domain.Clinic;
+import ua.su.repository.ClinicRepository;
+import ua.su.repository.MedicalProcedureRepository;
+import ua.su.service.FindSomethingByCriteria;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class FindSomethingInClinic2 extends ClinicServiceImpl implements FindSomethingByCriteria {
```

Comment:
It is better to rename it to "ClinicServiceSearchStreamAapi" it is going to be more detail described

Reply:
Thanks! The class was renamed from 'FindSomethingInClinic2.java' to 'ClinicServiceSearchStreamApi'.